spec/lib/notification_manager_spec.rb

```diff
@@ -66,17 +66,13 @@
       described_class.error(exception, message, ScienceWireClient.new)
     end
   end
-  context '.wos_logger' do
+  context 'WebOfScience.logger' do
     let(:wos_client) { WebOfScience::Client.new(Settings.WOS.AUTH_CODE) }
 
-    before do
-      described_class.class_variable_set(:@@wos_logger, nil)
-    end
-    it 'creates a single logger' do
-      expect(Logger).to receive(:new).with(Settings.WOS.LOG).once
-      described_class.error(exception, message, wos_client)
-    end
+    before { WebOfScience.class_variable_set(:@@logger, nil) }
+
     it 'logs errors' do
+      expect(WebOfScience).to receive(:logger).and_return(null_logger)
       expect(null_logger).to receive(:error).exactly(3)
       described_class.error(exception, message, wos_client)
     end
```

Comment:
Replace
- `expect(WebOfScience).to receive(:logger).and_return(null_logger)`
- with `expect(Logger).to receive(:new).with(Settings.WOS.LOG).once.and_return(null_logger)`

Not sure why the `null_logger` is receiving 3 errors, is that right?

Reply:
Nope.  That is 100% NOT the concern of `NotificationManager` anymore.

Reply:
Huh?  I'm not talking about `NotificationManager`, right.

Reply:
BTW, I think it gets 3 errors because there are 3 retries in the Faraday client.

Reply:
I think that is right.  In any case, the number of times is unchanged.